### PR TITLE
Documentation update on topic of drawing conflicts management in extension and extensions migration to use updated layout properly

### PR DESCRIPTION
**Pull request by alexander-fedorenko**
## Description
This PR brings new chapter to the Developer Guide / Extensions with a short explanation on how to deal with drawing conflicts that may occur.

**Please check if the PR fulfills these requirements**
- [ ] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [ ] Bugfix
 - [ ] Feature
 - [ ] Code style update (formatting, local variables)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build related changes
 - [ ] CI related changes
 - [x] Other... Please describe: Documentation change

## Issue

**What is the current behavior?**
Missing info on how to resolve drawing conflicts

**What is the new behavior?**
As per description

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

### CODE REVIEW

File: docs/developer-guide/extensions.md [hunk 1]
```diff
@@ -172,3 +172,115 @@ ConfigUtils.setConfigProp("extensionsFolder", "rest/config/loadasset");
...
+with this code located in extension's epics your tool `yourToolName` will be closed whenever: 
+- feature editor is open
+- another plugin or extension starts drawing. 
+
+"shutdownToolOnAnotherToolDrawing" supports passing custom callback to determine whether your tool is active (to prevent garbage action dispatching if it's already off) and custom callback to list actions to be dispatched.
+
+
+## Using "ResponsiveContainer" for dock panels
+
+Recent MapStore version (2022.01.xx) introduced layout improvements that apart from the other changes brings new sidebar menu
```
Comment (tdipisa) [inline docs/developer-guide/extensions.md:214]: not 2022.01.xx but 2022.02.00 so:

Starting with MapStore v2022.02.00, layout improvements have been introduced which, in addition to other changes, introduce a new sidebar menu

**Review by tdipisa — CHANGES_REQUESTED** [review]

**Review by tdipisa — CHANGES_REQUESTED** [review]
@alexander-fedorenko we can also consider to put a link in the migration guidelines to reach this section and highlight it to devs

**Comment by tdipisa** [comment]
> @alexander-fedorenko This part of the doc is fine for me but I don't see the part related to panel registration to make them exclusive. I remember in the PR that you could register a panel to make it close if another one is opening but maybe I'm wrong

Good point. Thank you @allyoucanmap. Let us know for all the above @alexander-fedorenko, please.

**Comment by alexander-fedorenko** [comment]
@allyoucanmap you're right, chapter about adding dock panels into list is missing. I will add it along with the reference to the updated page in migration guidelines asap.

**Review by tdipisa — APPROVED** [review]

**Review by allyoucanmap — APPROVED** [review]

**Pull request merged**